Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py

```diff
@@ -850,6 +863,152 @@ def set_blob_properties_command(client: Client, args: Dict[str, Any]) -> Command
     return command_results
 
 
+# generate signature helper function
+def generate_sas_signature(account_key: str, cr: str, sp: str, signedstart: str, expiry: str, sr: str, api_version: str, sip: str = ''):
+    """
+    Generate sas token for Container
+
+    Args:
+        account_key: account key of conatiner.
+        cr: canonicalizedResource.
+        sp:SignedPermissions.
+        signedstart: start time for sas token.
+        expiry : Expiry time for sas token.
+    Returns:
+        sas token
+
+    """
+    if sip:
+        StringToSign = (sp + "\n" +
+                        signedstart + "\n"
+                        + expiry + "\n"
+                        + cr + "\n"
+                        + "" + "\n"
+                        + sip + "\n"
+                        + "https" + "\n"
+                        + api_version + "\n"
+                        + sr + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "").encode('UTF-8')
+        signed_hmac_sha256 = hmac.new(base64.b64decode(account_key), StringToSign, hashlib.sha256)
+        sig = base64.b64encode(signed_hmac_sha256.digest())
+
+        token = {
+            'sp': sp,
+            'st': signedstart,
+            'se': expiry,
+            'sip': sip,
+            'spr': "https",
+            'sv': api_version,
+            'sr': sr,
+            'sig': sig
+        }
+    else:
+        StringToSign = (sp + "\n" +
+                        signedstart + "\n"
+                        + expiry + "\n"
+                        + cr + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "https" + "\n"
+                        + api_version + "\n"
+                        + sr + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "").encode('UTF-8')
+        signed_hmac_sha256 = hmac.new(base64.b64decode(account_key), StringToSign, hashlib.sha256)
+        sig = base64.b64encode(signed_hmac_sha256.digest())
+        token = {
+            'sp': sp,
+            'st': signedstart,
+            'se': expiry,
+            'spr': "https",
+            'sv': api_version,
+            'sr': sr,
+            'sig': sig
+        }
```

Comment:
```suggestion
   string_to_sign = (sp + "\n" +
                  signedstart + "\n"
                  + expiry + "\n"
                  + cr + "\n"
                  + "" + "\n"
                  + sip + "\n"
                  + "https" + "\n"
                  + api_version + "\n"
                  + sr + "\n"
                  + "" + "\n"
                  + "" + "\n"
                  + "" + "\n"
                  + "" + "\n"
                  + "" + "\n"
                  + "").encode('UTF-8')
signed_hmac_sha256 = hmac.new(base64.b64decode(account_key), string_to_sign, hashlib.sha256)
sig = base64.b64encode(signed_hmac_sha256.digest())

token = {
    'sp': sp,
    'st': signedstart,
    'se': expiry,
    'sip': sip,
    'spr': "https",
    'sv': api_version,
    'sr': sr,
    'sig': sig
}
if sip:
    token['sip'] = sip
```

We don't need to duplicate code, it's bassically the same code but yo just need to add sip to the token if it exists